cmd/keeper/cmd/keeper.go

```diff
@@ -1588,6 +1589,17 @@ func (p *PostgresKeeper) postgresKeeperSM(pctx context.Context) {
 		if err := pgm.Reload(); err != nil {
 			log.Errorw("failed to reload postgres instance", err)
 		}
+		needsRestart, err := pgm.IsRestartRequired(changedParams)
+		if err != nil {
+			log.Errorw("Failed to checked if restart is required", err)
+		}
+
+		if needsRestart {
```

Comment:
I'll make this opt in adding an option in the cluster spec. I'll leave it disabled by default (at least for the moment, to not break current expected behavior and avoid possible issue when users provides a wrong parameter that will break instance restart)

Reply:
@sgotti As you suggested, We will make the pgRestart optional(disbaled by default) and add some integration tests. We will try to close this by this weekend.
cc @aswinkarthik